Sources/MongoSwift/ClientSession.swift

```diff
@@ -143,18 +144,28 @@ public final class ClientSession {
         return connection
     }
 
-    /// Destroy the underlying `mongoc_client_session_t` and ends this session. Has no effect if this session is
-    /// already ended.
-    internal func end() {
-        if case let .started(session, _) = self.state {
-            mongoc_client_session_destroy(session)
+    /// Ends this `ClientSession`. Call this method when you are finished using the session. You must ensure that all
+    /// operations using this session have completed before calling this. The returned future must be fulfilled before
+    /// the session goes out of scope.
```

Comment:
good point. I think users do need to ensure the future is fulfilled before the parent client is closed though, otherwise the session's connection might not be returned to the pool before its destroyed. I updated the docstring to reflect this.
I also updated `MongoClient.close()` with some similar language